packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/internals/internal_bulk_resolve.test.ts

```diff
@@ -368,89 +368,37 @@ describe('internalBulkResolve', () => {
       ]);
     });
 
-    it('migrates the resolved objects', async () => {
+    // TODO
+    it('migrates and decrypts the resolved objects', async () => {
       const objects = [
         { type: OBJ_TYPE, id: '1' },
         { type: OBJ_TYPE, id: '2' },
       ];
       const params = setup(objects, { namespace });
       mockBulkResults({ found: false }, { found: false });
       mockMgetResults({ found: true }, { found: true });
-      migrator.migrateDocument.mockImplementation(
-        (doc) => `migrated-${doc}` as unknown as SavedObjectUnsanitizedDoc
+      const migrationHelper = apiContext.helpers.migration;
+      migrationHelper.migrateAndDecryptStorageDocument.mockImplementation(({ document }) =>
+        Promise.resolve(`migrated-${document}` as unknown as SavedObject)
       );
 
-      await expect(internalBulkResolve(params)).resolves.toHaveProperty('resolved_objects', [
-        expect.objectContaining({ saved_object: 'migrated-mock-obj-for-1' }),
-        expect.objectContaining({ saved_object: 'migrated-mock-obj-for-2' }),
-      ]);
-
-      expect(migrator.migrateDocument).toHaveBeenCalledTimes(2);
-      expect(migrator.migrateDocument).nthCalledWith(
-        1,
-        'mock-obj-for-1',
-        expect.objectContaining({ allowDowngrade: expect.any(Boolean) })
+      await expect(internalBulkResolve(params, apiContext)).resolves.toHaveProperty(
+        'resolved_objects',
+        [
+          expect.objectContaining({ saved_object: 'migrated-mock-obj-for-1' }),
+          expect.objectContaining({ saved_object: 'migrated-mock-obj-for-2' }),
+        ]
       );
-      expect(migrator.migrateDocument).nthCalledWith(2, 'mock-obj-for-2', {
-        allowDowngrade: expect.any(Boolean),
-      });
-    });
-  });
-
-  describe('with encryption extension', () => {
-    const namespace = 'foo';
-
-    const attributes = {
-      attrNotSoSecret: '*not-so-secret*',
-      attrOne: 'one',
-      attrSecret: '*secret*',
-      attrThree: 'three',
-      title: 'Testing',
-    };
 
-    beforeEach(() => {
-      mockGetSavedObjectFromSource.mockImplementation((_registry, type, id) => {
-        return {
-          id,
-          type,
-          namespaces: [namespace],
-          attributes,
-          references: [],
-        } as SavedObject;
+      expect(migrationHelper.migrateAndDecryptStorageDocument).toHaveBeenCalledTimes(2);
+      expect(migrationHelper.migrateAndDecryptStorageDocument).nthCalledWith(1, {
+        document: 'mock-obj-for-1',
+        typeMap: undefined,
+      });
+      expect(migrationHelper.migrateAndDecryptStorageDocument).nthCalledWith(2, {
+        document: 'mock-obj-for-2',
+        typeMap: undefined,
       });
-    });
-
-    it('only attempts to decrypt and strip attributes for types that are encryptable', async () => {
```

Comment:
I think it's ok to remove this test. The lower level logic is sufficiently tested in the encryption extension unit tests. If we feel it's necessary we can add a unit test for the SOR encryption helper.